pkg/providers/vsphere/template.go

```diff
@@ -172,7 +172,8 @@ func buildTemplateMapCP(
 		"controlPlaneVsphereResourcePool":      controlPlaneMachineSpec.ResourcePool,
 		"vsphereServer":                        datacenterSpec.Server,
 		"controlPlaneVsphereStoragePolicyName": controlPlaneMachineSpec.StoragePolicyName,
-		"vsphereTemplate":                      controlPlaneMachineSpec.Template,
+		"controlPlaneTemplate":                 controlPlaneMachineSpec.Template,
```

Comment:
If we are making these changes, can we change the reference of the worker node vsphereTemplate to `workerTemplate`?